matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/handler/SyncResponsePostTreatmentAggregatorHandler.kt

```diff
@@ -68,4 +78,22 @@ internal class SyncResponsePostTreatmentAggregatorHandler @Inject constructor(
             updateUserAccountDataTask.execute(UpdateUserAccountDataTask.DirectChatParams(directMessages = directChats))
         }
     }
+
+    private suspend fun fetchAndUpdateUsers(usersToFetch: MutableList<String>) {
+        val userProfiles = fetchUsers(usersToFetch)
+        userProfiles.forEach { saveUserLocally(monarchy, it) }
+        usersToFetch.clear()
```

Comment:
This is probably not necessary to clear the list.

Reply:
If we don't though, wouldn't we keep fetching the same users even when we don't need to fetch them anymore? I think it could increase bandwith and performance usage the longer the app is kept on

Reply:
Nope. A brand new `SyncResponsePostTreatmentAggregator` is created for each sync request. See https://github.com/vector-im/element-android/blob/develop/matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/SyncResponseHandler.kt#L103

Reply:
`usersToFetch` parameter could probably be of type `List` (instead of `MutableList`)

Reply:
Ahhh this wasn't clear to me at first. Changes pushed